src/ar/cimaleek/src/eu/kanade/tachiyomi/animeextension/ar/cimaleek/interceptor/GetSourcesInterceptor.kt

```diff
@@ -0,0 +1,99 @@
+package eu.kanade.tachiyomi.animeextension.ar.cimaleek.interceptor
+
+import android.annotation.SuppressLint
+import android.app.Application
+import android.os.Handler
+import android.os.Looper
+import android.webkit.WebResourceRequest
+import android.webkit.WebResourceResponse
+import android.webkit.WebSettings
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import eu.kanade.tachiyomi.network.GET
+import okhttp3.Headers
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.io.IOException
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+class GetSourcesInterceptor(private val searchRegex: Regex, private val globalHeaders: Headers) : Interceptor {
+    private val context = Injekt.get<Application>()
+    private val handler by lazy { Handler(Looper.getMainLooper()) }
+
+    private val initWebView by lazy {
+        WebSettings.getDefaultUserAgent(context)
+    }
+
+    @Synchronized
+    override fun intercept(chain: Interceptor.Chain): Response {
+        initWebView
+
+        val request = chain.request()
+
+        try {
+            val newRequest = resolveWithWebView(request)
+
+            return chain.proceed(newRequest ?: request)
+        } catch (e: Exception) {
+            throw IOException(e)
+        }
+    }
+
+    @SuppressLint("SetJavaScriptEnabled")
+    private fun resolveWithWebView(request: Request): Request? {
```

Comment:
Wouldn't it be better to return the string of the url instead of a request? It wouldn't be an extractor after that and then you'd be able to use playlist-utils instead, simplifying some of the process

Reply:
hmm let me try

Reply:
but can i override intercept function to return string instead of response ?? @Secozzi

Reply:
It shouldn't be an interceptor, just call it an extractor instead

Reply:
what exactly ? i mean in Interceptor Class the intercept function returns Response i can't override that to return a String instead
i use interceptor cuz i can't make and extractor

Reply:
You shouldn't inherit from interceptor in the first place

Reply:
O.o i totally don't get it, i mean then how am i supposed to intercept the requests ??

Reply:
You load the URL in webview like you did before